dds/DCPS/TypeAssignability.cpp

```diff
@@ -0,0 +1,2083 @@
+      } else { // Add a key designator to each member
+        for (size_t i = 0; i < mseq.members.size(); ++i) {
+          MemberFlag flags = mseq.members[i].common.member_flags;
+          if ((flags & IS_KEY) != IS_KEY) {
+            mseq.members[i].common.member_flags |= IS_KEY;
+          }
+        }
+      }
+    } else if (TK_UNION == type.kind) {
+      if ((type.union_type.discriminator.common.member_flags & IS_KEY) == IS_KEY) {
+        // Remove all non-key members
+        type.union_type.member_seq = Sequence<MinimalUnionMember>();
+      }
+    } else if (TK_ALIAS == type.kind) {
+      const TypeIdentifier& base = get_base_type(type);
+      MinimalTypeObject base_obj = typelookup_.lookup_minimal(base);
+      hold_key(base_obj);
+      type = base_obj;
+    }
+  }
+
+  /**
+   * @brief The input must be of type TK_ALIAS
+   * Return the non-alias base type identifier of the input
+   */
+  const TypeIdentifier& TypeAssignability::get_base_type(const MinimalTypeObject&
+                                                         type) const
+  {
+    const TypeIdentifier& base = *type.alias_type.body.common.related_type.in();
+    switch (base.kind) {
+    case EK_COMPLETE:
+    case EK_MINIMAL: {
+      const MinimalTypeObject& type_obj = typelookup_.lookup_minimal(base);
+      if (TK_ALIAS == type_obj.kind) {
+        return get_base_type(type_obj);
+      }
+      return base;
+    }
+    default:
+      return base;
+    }
+  }
+
+  /**
+   * @brief The first argument must be TK_ENUM and is the type object
+   * of a key member of the containing struct. Therefore, there must be a
+   * member with the same ID (and name) in the other struct type.
+   */
+  bool TypeAssignability::struct_rule_enum_key(const MinimalTypeObject& tb,
+                                               const CommonStructMember& ma) const
+  {
+    if (EK_MINIMAL != ma.member_type_id->kind) {
+      return false;
+    }
+
+    const MinimalEnumeratedLiteralSeq& literals_b = tb.enumerated_type.literal_seq;
+    const MinimalTypeObject&
+      toa = typelookup_.lookup_minimal(*ma.member_type_id.in());
+    const MinimalEnumeratedLiteralSeq* literals_a = 0;
+    if (TK_ENUM == toa.kind) {
+      literals_a = &toa.enumerated_type.literal_seq;
+    } else if (TK_ALIAS == toa.kind) {
+      const TypeIdentifier& base_a = get_base_type(toa);
+      if (EK_MINIMAL == base_a.kind) {
+        const MinimalTypeObject& base_obj_a = typelookup_.lookup_minimal(base_a);
+        if (TK_ENUM == base_obj_a.kind) {
+          literals_a = &base_obj_a.enumerated_type.literal_seq;
+        } else {
+          return false;
+        }
+      } else {
+        return false;
+      }
+    } else {
+      return false;
+    }
+
+    // All literals in tb must appear as literals in toa
+    for (size_t j = 0; j < literals_b.members.size(); ++j) {
+      const NameHash& h_b = literals_b.members[j].detail.name_hash;
+      ACE_CDR::ULong
```

Comment:
Try to avoid splitting the variable's type and name on separate lines.  We are not using a strict 80-column limit, but it's also good to keep lines at a reasonable length -- so these two style guidelines are somewhat at odds, but it usually works out OK.  One thing that will save 2 characters is that the style guide says that the method itself starts at 0 indent.

Reply:
Updated to avoid line splitting.